### PR TITLE
add github action, changelog, docs

### LINKED ISSUE #95: Document release process (and cut 0.4.1)
It would be great to get a new release (0.4.1) out. @martindurant - I'd be happy to add a small section in the docs regarding your preferred way to do this. 

Or we could add a ["publish" github action](https://github.com/intake/intake-stac/blob/master/.github/workflows/pypipublish.yaml) to push to pypi as we're doing in intake-stac that just requires pushing a new tag. A changelog would also be nice to add.

### PR DESCRIPTION
**Pull request by scottyhq**
fixes #95 

contributing.rst and changelog model copied from [intake-stac](https://github.com/intake/intake-stac)

### CODE REVIEW

File: .github/workflows/pypipublish.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools setuptools-scm wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (scottyhq) [inline .github/workflows/pypipublish.yaml:23]: @martindurant - this workflow requires you to add secrets either in this repository specifically, or for the `intake` organization https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-organization

Comment (martindurant) [reply .github/workflows/pypipublish.yaml:23]: I'd love to have the workflow eventually for all the repos, so I'll add it to the org.

**Comment by martindurant** [comment]
Thanks for putting this together!

File: .github/workflows/pypipublish.yaml (hunk 1 above)
Comment (martindurant) [reply .github/workflows/pypipublish.yaml:23]: Added. This means that you should be able to merge and tag, and we'll get a new version. The resultant conda-forge PR will still need merging when it's ready.

**Comment by scottyhq** [comment]
great! i put this together rather hastily so i'll look it over once more later today or tomorrow before merging

**Pull request merged**